lib/spack/spack/test/cmd/config.py

```diff
@@ -645,3 +648,49 @@ def check_config_updated(data):
     assert isinstance(data['install_tree'], dict)
     assert data['install_tree']['root'] == '/fake/path'
     assert data['install_tree']['projections'] == {'all': '{name}-{version}'}
+
+
+def test_config_prefer_upstream(tmpdir_factory, install_mockery, mock_fetch,
```

Comment:
This test needs the `mutable_config` fixture so that failures won't leave detritus in real config directories